jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java

```diff
@@ -2426,7 +2426,10 @@ protected PdfObject getObject (int objIndex, int recGuard)
                     }
                 }
                 /* And finally extract the object from the object stream. */
-                return ostrm.getObject (objIndex);
+                /* Need to ensure the object number is set */
+                PdfObject obj = ostrm.getObject (objIndex);
+                obj.setObjNumber (objIndex);
```

Comment:
Setting the object number can safely be moved into the ObjectStream.getObject() method for the benefit of any other callers.